core/vm/evm.go

```diff
@@ -65,6 +66,12 @@ func (evm *EVM) precompile(addr common.Address) (StatefulPrecompiledContract, bo
 	default:
 		precompiles = PrecompiledContractsHomestead
 	}
+
+	// Enable the optional stateful precompiles
+	if evm.chainRules.IsAllowListEnabled {
+		precompiles[precompile.ModifyAllowListAddress] = precompile.ModifyAllowListPrecompile
+		precompiles[precompile.ReadAllowListAddress] = precompile.ReadAllowListPrecompile
```

Comment:
Why have a separate precompile for reading? Is this to avoid more complex input parsing? (will continue reading)

Reply:
Seems like we could instead use the first byte as a function selector? like 0x0 == modify and 0x1 == read

I think that is a much more scalable pattern than splitting.